File: entrega_5/Java/src/topology/TopologyFactory.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package topology;

/**
 *
 * @author Geraldo e Iago
 */
public class TopologyFactory
{
    private ITopology top;
    public TopologyFactory(TopologyEnum tEnum)
    {
        switch(tEnum)
        {
            default:
            case STAR:
                this.top = new TopologyStar();
                break;
        }
    }
    
    public ITopology get()
    {
        return this.top;
    }
}

File: entrega_1/pso_simples/topology/topologyfactory.cpp
#include "topologyfactory.h"

TopologyFactory::TopologyFactory(TopologyFactory::TOPOLOGY top)
{
    switch (top) {
    default:
    case TOPOLOGY::STAR:
        this->topology = new TopologyStar();
        break;
    case TOPOLOGY::MASTER_SLAVE:
        this->topology = new TopologyMasterSlave();
        break;
    case TOPOLOGY::RING:
        this->topology = new TopologyRing();
        break;
    }
}

TopologyFactory::~TopologyFactory()
{

}

ITopology *TopologyFactory::get()
{
    return this->topology;
}
